partiql-ast/src/main/java/org/partiql/ast/literal/LiteralInt.java

```diff
@@ -0,0 +1,16 @@
+package org.partiql.ast.literal;
+
+import lombok.EqualsAndHashCode;
+
+/**
+ * TODO docs
+ */
+@EqualsAndHashCode(callSuper = false)
+public class LiteralInt extends Literal {
+    public int value;
+
+    public LiteralInt(int value) {
+        super(String.format("%d", value));
+        this.value = value;
+    }
+}
```

Comment:
I don't believe there should be both a LiteralInt and LiteralLong, there should be one "integral literal" which may just be called LiteralInteger.

Reply:
To be honest, it would be nice to just have LiteralNumber which has the necessary APIs / details to model floating point, decimals, and integers. But also there are places in the BNF where we might want to enforce exact numeric or integral.

Reply:
Yeah -- the literals, especially the numerics, had a lot of different ways to model. I played around w/ some different approaches but wasn't yet sure of which stage (i.e. parser, ast, planner) should perform the validation and conversion of the literal text.

I'll sync offline w/ you and @johnedquinn on where the text -> value conversion should take place and where to perform any validation.